auth.go

```diff
@@ -418,6 +418,17 @@ func createRequestBody(sc *snowflakeConn, sessionParameters map[string]interface
 		requestMain.Authenticator = AuthTypeOAuth.String()
 		requestMain.Token = sc.cfg.Token
 	case AuthTypeOkta:
+		samlResponse, err := authenticateBySAML(
```

Comment:
Let me understand it correctly. So the flow is that first we call identity provider and it returns some SAML response. Then we take response and use it to call Snowflake. The problem is that when retry happens the same SAML response cannot be used to call Snowflake, am I right? So when we create a body, we have to regenerate SAML response every time?

Does the same problem happens for authenticating using external browsers?

Reply:
Yes, we will need to regenerate SAML response when retrying. The ticket is logged for okta only.